rustls/src/msgs/handshake.rs

```diff
@@ -1619,19 +1829,25 @@ impl Codec<'_> for ServerKeyExchangePayload {
 
 impl ServerKeyExchangePayload {
     #[cfg(feature = "tls12")]
-    pub(crate) fn unwrap_given_kxa(
-        &self,
-        kxa: KeyExchangeAlgorithm,
-    ) -> Option<EcdheServerKeyExchange> {
+    pub(crate) fn unwrap_given_kxa(&self, kxa: KeyExchangeAlgorithm) -> Option<ServerKeyExchange> {
         if let Self::Unknown(ref unk) = *self {
             let mut rd = Reader::init(unk.bytes());
 
-            let result = match kxa {
-                KeyExchangeAlgorithm::ECDHE => EcdheServerKeyExchange::read(&mut rd),
+            let server_kx_params = match kxa {
+                KeyExchangeAlgorithm::ECDHE => {
+                    ServerKeyExchangeParams::Ecdh(ServerEcdhParams::read(&mut rd).ok()?)
+                }
+                KeyExchangeAlgorithm::DHE => {
+                    ServerKeyExchangeParams::Dh(ServerDhParams::read(&mut rd).ok()?)
+                }
+            };
+            let result = ServerKeyExchange {
+                params: server_kx_params,
+                dss: DigitallySignedStruct::read(&mut rd).ok()?,
             };
 
             if !rd.any_left() {
-                return result.ok();
+                return Some(result);
```

Comment:
Style nit: would suggest inlining the binding here.

Reply:
Not sure how to do that here.

Reply:
Ahh, you're right. Never mind!